### PR TITLE
Customize commit messages

### PR DESCRIPTION
**Pull request by paulschuberth**
This change adds a new input to the action. It can be used to customize
the message of the commit created by this action.

This input is not required and has the default value `Update Gradle
Wrapper from %sourceVersion% to %targetVersion%`, where `%sourceVersion%`
will be replaced by the current/old version of the Gradle Wrapper and
`%targetVersion%` will be replaced by the new version of the Gradle
Wrapper.

The PR title and body remain unaffected by this change.

Implements #246

### CODE REVIEW

**Review by ChrisCarini — APPROVED** [review]
LGTM!

**Comment by paulschuberth** [comment]
Thanks for the review @ChrisCarini. I've rebased my branches onto main to resolve the conflicts.

**Review by ChrisCarini — APPROVED** [review]
LGTM! @cristiangreco , thoughts? 😄

**Comment by paulschuberth** [comment]
Hi @cristiangreco, I hope all is well 😊

Just checking if there is anything I can improve here to get this merged. I would appreciate any feedback.

If you just haven't gotten around to take a look that's of course fine too.

**Comment by Fs02** [comment]
> The PR title and body remain unaffected by this change.

will be very useful for me if this also supports PR title

**Comment by paulschuberth** [comment]
> > The PR title and body remain unaffected by this change.
> 
> 
> 
> will be very useful for me if this also supports PR title

I can look into that over the weekend. It might become it's own PR though, to keep the changes limited in their scopes.

**Comment by ChrisCarini** [comment]
> > > The PR title and body remain unaffected by this change.
> > 
> > 
> > will be very useful for me if this also supports PR title
> 
> I can look into that over the weekend. It might become it's own PR though, to keep the changes limited in their scopes.

FWIW (I'm not the owner of this repo), it'd make sense to do that in a separate change. I personally think *this* change is ready for an approval and to be merged+released, so I wouldn't want to delay that on a separate (although, very related) feature.

@cristiangreco - thoughts? Think we can get this PR merged in and released to folks soon?

**Review by ChrisCarini — APPROVED** [review]
LGTM! @cristiangreco mind taking a few minutes to review this when you have a moment?

**Comment by paulschuberth** [comment]
As follow up to https://github.com/gradle-update/update-gradle-wrapper-action/pull/695#discussion_r1240984905 this PR needs some adjustments. I am about to leave for a short vacation and can tackle it some time next week.

**Comment by paulschuberth** [comment]
Updated this PR to match the nullability behaviour of #695

**Review by ChrisCarini — APPROVED** [review]

**Review by ChrisCarini — APPROVED** [review]

**Review by ChrisCarini — APPROVED** [review]

**Comment by ChrisCarini** [comment]
@cristiangreco - are you able to take a look at this PR that @paulschuberth has put forth when you have a moment?

**Comment by paulschuberth** [comment]
Hi @cristiangreco 
Is there still something that's blocking the merge? Something to improve or change on my end?

**Comment by ChrisCarini** [comment]
@cristiangreco - are you able to take a look at this PR that @paulschuberth has put forth when you have a moment?

**Comment by ChrisCarini** [comment]
Hey @cristiangreco - any chance you could take a look at this PR when you have a moment? 🙏 I'd love to see it merged in.

**Comment by ChrisCarini** [comment]
Hi @cristiangreco - I'm sure you're very busy with other work, but when you have ~5 min, would you mind taking another peak at this PR that @paulschuberth put forward? It's been open ~14 months now, and I'd love to see it reviewed / merged in, if you feel it's ready!

File: action.yml [hunk 1]
```diff
@@ -56,6 +56,14 @@ inputs:
       used to replace the source version placeholder.
     required: false
     default: 'Update Gradle Wrapper from %sourceVersion% to %targetVersion%'
+  commit-message-template:
+    description: |
+      Template used for commit message
```
Comment (cristiangreco) [inline action.yml:61]:
```suggestion
      Template used for commit message.
```

**Pull request merged**